src/kiwixapp.cpp

```diff
@@ -577,3 +592,35 @@ void KiwixApp::saveCurrentTabIndex()
 {
   return mp_session->setValue("currentTabIndex", getTabWidget()->currentIndex());
 }
+
+void KiwixApp::addRemovedZimBookInfo(const QList<kiwix::Book> &books)
+{
+    QMutexLocker locker(&m_updateBookInfoMutex);
+    auto bookInfoMap = mp_session->value("removedZimBookInfo", QVariantMap{}).toMap();
+    for (auto& book : books)
+    {
+        auto id = QString::fromStdString(book.getId());
+        auto name = QString::fromStdString(book.getName());
+        auto path = QString::fromStdString(book.getPath());
+        bookInfoMap[id] = QJsonObject{{"name", name}, {"path", path}};
+    }
+    mp_session->setValue("removedZimBookInfo", bookInfoMap);
+}
+
+/**
+ * @brief Removes only the book infos that has no trace left in the app.
+ * 
+ */
+void KiwixApp::cleanupRemovedZimBookInfo()
+{
+    QMutexLocker locker(&m_updateBookInfoMutex);
+    auto bookInfoMap = mp_session->value("removedZimBookInfo", QVariantMap{}).toMap();
+    auto existingZimIds = getTabWidget()->getTabZimIds();
+
+    for (const auto& zimId : bookInfoMap.keys())
+    {
+        if (!existingZimIds.contains(zimId))
+            bookInfoMap.remove(zimId);
+    }
+    mp_session->setValue("removedZimBookInfo", bookInfoMap);
+}
```

Comment:
To me something seems wrong about saving this information in the session. Please give me some time to come up with a better solution or admit that none exists.